src/GuzzleRequestAdapter.php

```diff
@@ -281,152 +368,320 @@ public function getBaseUrl(): string
      * Create and returns a PSR 7 Request object from {@link RequestInformation}
      *
      * @param RequestInformation $requestInformation
+     * @param SpanInterface|null $span
      * @return RequestInterface
      * @throws UriException
      */
-    public function getPsrRequestFromRequestInformation(RequestInformation $requestInformation): RequestInterface
+    public function getPsrRequestFromRequestInformation(RequestInformation $requestInformation,
+                                                        ?SpanInterface $span = null): RequestInterface
     {
-        $requestInformation->pathParameters["baseurl"] = $this->getBaseUrl();
-        return new Request(
-            $requestInformation->httpMethod,
-            $requestInformation->getUri(),
-            $requestInformation->getHeaders()->getAll(),
-            $requestInformation->content
-        );
+        $span = $span ?? $this->tracer->spanBuilder('getHttpResponseMessage')
+            ->startSpan();
+        $currentContext = Context::getCurrent();
+        $psrRequestFromInfoSpan = $this->tracer->spanBuilder('getPsrRequestFromRequestInformation')
+            ->setParent($currentContext)->addLink($span->getContext())
+            ->startSpan();
```

Comment:
If no parent span is passed, we can initialise the psrRequestInfoSpan without a parent reference